beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -255,6 +256,8 @@ func (s *Service) clearPendingSlots() {
 // Delete block from the list from the pending queue using the slot as key.
 // Note: this helper is not thread safe.
 func (s *Service) deleteBlockFromPendingQueue(slot uint64, b *ethpb.SignedBeaconBlock, r [32]byte) {
+	mutexasserts.AssertRWMutexLocked(&s.pendingQueueLock)
```

Comment:
Using assert here since this method doesn't return an error. 

Please note that AssertRWMutexLocked does nothing if not built with `gotags=debug`. I've set E2E to run with this gotag, so we can attempt to catch any violates in e2e.